stan/math/prim/scal/err/throw_domain_error_vec.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_THROW_DOMAIN_ERROR_VEC_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_THROW_DOMAIN_ERROR_VEC_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/scal/err/throw_domain_error.hpp>
+#include <sstream>
+#include <string>
+
+namespace stan {
+namespace math {
+
+/**
+ * Throw a domain error with a consistently formatted message.
```

Comment:
Thanks also for writing such clear and thorough doc.

I think Boost also throws domain errors directly from some of its functions, but I'm not 100% sure about that.

Reply:
I have copied what was there!